### PR TITLE
1472 tag replication

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: app/models/replication/obj_proxy.rb [hunk 1]
```diff
@@ -156,34 +162,36 @@ def get_copy_ancestry(context)
...
 
-  def standardizable_col_mappings(replicator, context)
+  def standardizable_col_mappings(replicator, _context)
     mappings = []
     case replicator.mode
     when :promote
       mappings << ["is_standard", true] if klass.standardizable?
     when :clone
-      mappings << "mission_id"
+      mappings << "mission_id" if klass_has_mission_id
```
Comment (DevneyHamilton) [inline app/models/replication/obj_proxy.rb]: Using `klass_has_mission_id` here is needed for `Tagging` which does not have mission id.

File: app/models/replication/obj_proxy.rb [hunk 2]
```diff
@@ -156,34 +162,36 @@ def get_copy_ancestry(context)
...
       mappings << "is_standard" if klass.standardizable?
     when :to_mission
-      mappings << ["mission_id", quote_or_null(replicator.target_mission_id)]
+      mappings << ["mission_id", quote_or_null(replicator.target_mission_id)] if klass_has_mission_id
```
Comment (DevneyHamilton) [inline app/models/replication/obj_proxy.rb]: Using `klass_has_mission_id` here is needed for `Tagging` which does not have mission id.

File: app/models/question.rb [hunk 3]
```diff
@@ -40,66 +47,68 @@ class Question < ApplicationRecord
...
+    :first_level_options,
+    :multilevel?,
+    :level_count,
+    :level,
+    :levels,
+    :sms_formatting_as_text?,
+    :sms_formatting_as_appendix?,
+    to: :option_set, allow_nil: true
+
+  replicable child_assocs: %i[option_set taggings], backwards_assocs: :questioning, sync: :code,
```
Comment (DevneyHamilton) [inline app/models/question.rb]: `replicable` declaration updated to add `taggings`. all other changes in this file are rubocop fixes.

File: app/models/replication/replicable.rb [hunk 4]
```diff
@@ -1,19 +1,23 @@
-# Behaviors that handle replicating creation and updates to copies of core objects (forms, questions, etc.) within and across missions.
```
Comment (DevneyHamilton) [inline app/models/replication/replicable.rb]: This file only has rubocop fixes and a comment update on `:backward_assocs`. No functionality changes for tag replication.

File: spec/models/replication/tag_replication_spec.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  describe "to_mission" do
+    let(:orig_mission) { nil }
+    let(:copy_q) { orig_q.replicate(mode: :to_mission, dest_mission: mission) }
+
+    context "basic" do
+      it "should replicate tag when replicates a standard library question to a mission" do
+        orig_q.reload
+        expect(orig_q.id).not_to eq(copy_q.id)
+        expect(copy_q.tags.count).to eq(2)
+        [orig_tag1, orig_tag2].each_with_index do |orig_tag, _i|
```
Comment (smoyth) [inline spec/models/replication/tag_replication_spec.rb]: s/each_with_index/each/ ??

File: spec/models/replication/tag_replication_spec.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        new_tag = find_tag_by_name(copy_q, orig_tag.name)
+        expect(new_tag.id).to eq(orig_tag.id)
+        expect(new_tag.name).to eq(orig_tag.name)
+        expect(new_tag.mission).to eq(mission)
+      end
+    end
+  end
+
+  def find_tag_by_name(question, name)
+    question.tags.detect { |t| t.name == name }
```
Comment (smoyth) [inline spec/models/replication/tag_replication_spec.rb]: `question.tags.find_by(name: name)` will hit the DB and make extra sure the stuff is persisted.

**Comment by DevneyHamilton** [comment]
@smoyth I made the tweaks in the new spec file. The hound errors are holdovers from the autofixing on other files (existing code) that I didn't have time to address. Since the migration is a piece of work, leaving it as its own separate issue, 9126 for next sprint.

**Pull request merged**